### PR TITLE
chore: enable more BiDi tests

### PR DESCRIPTION
**Pull request by Lightning00Blade**

### CODE REVIEW

File: test/TestExpectations.json [hunk 1]
```diff
@@ -4372,5 +4366,101 @@
     "platforms": ["darwin", "linux", "win32"],
     "parameters": ["cdp", "chrome", "headless"],
     "expectations": ["FAIL", "PASS"]
+  },
+  {
+    "testIdPattern": "[frame.spec] Frame specs Frame Management should send events when frames are manipulated dynamically",
```
Comment (Lightning00Blade) [inline test/TestExpectations.json]: Intentionally not sorted so we can remove the  `chrome` if after Firefox passes as well.

File: packages/puppeteer-core/src/common/bidi/BrowsingContext.ts [hunk 2]
```diff
@@ -101,7 +101,7 @@ export class CDPSessionWrapper extends CDPSession {
 
   override async detach(): Promise<void> {
     cdpSessions.delete(this.id());
-    if (this.#context.supportsCDP()) {
+    if (!this.#detached && this.#context.supportsCDP()) {
```
Comment (Lightning00Blade) [inline packages/puppeteer-core/src/common/bidi/BrowsingContext.ts:104]: Causes issue as the target may have already been detached.

File: packages/puppeteer-core/src/common/bidi/Sandbox.ts [hunk 3]
```diff
@@ -68,6 +70,19 @@ export class Sandbox implements RealmBase {
   constructor(context: Realm, timeoutSettings: TimeoutSettings) {
     this.#realm = context;
     this.#timeoutSettings = timeoutSettings;
+
+    // TODO: we should track the correct realm
+    // Temporary fix
+    // This only works for the main context realm
+    this.#realm.on(Bidi.ChromiumBidi.Script.EventNames.RealmCreated, () => {
```
Comment (Lightning00Blade) [inline packages/puppeteer-core/src/common/bidi/Sandbox.ts]: There should be a better way to handle this, still looking over it, as it was part of an old stash,

File: packages/puppeteer-core/src/api/HTTPResponse.ts [hunk 4]
```diff
@@ -157,7 +157,7 @@ export class HTTPResponse {
    * True if the response was served by a service worker.
    */
   fromServiceWorker(): boolean {
```
Comment (OrKoN) [inline packages/puppeteer-core/src/api/HTTPResponse.ts]: should we mark it as abstract to make sure we don't accidentially start returning false for everything?

Comment (Lightning00Blade) [reply packages/puppeteer-core/src/api/HTTPResponse.ts]: Happy to refactor this in another PR as this class is not abstract. Only moving it into the BiDi class for now.

File: packages/puppeteer-core/src/common/bidi/Sandbox.ts (hunk 3 above)
Comment (Lightning00Blade) [reply packages/puppeteer-core/src/common/bidi/Sandbox.ts]: Left a comment, and we do the same thing as per PuppeteerUtils.

File: packages/puppeteer-core/src/common/Frame.ts [hunk 5]
```diff
@@ -63,7 +63,6 @@ export class Frame extends BaseFrame {
   _frameManager: FrameManager;
   override _id: string;
   _loaderId = '';
-  override _hasStartedLoading = false;
```
Comment (OrKoN) [inline packages/puppeteer-core/src/common/Frame.ts:66]: is it not used anymore?

**Review by OrKoN — APPROVED** [review]

File: packages/puppeteer-core/src/common/Frame.ts (hunk 5 above)
Comment (Lightning00Blade) [reply packages/puppeteer-core/src/common/Frame.ts:66]: This leave on the api Frame class and as a `"protected"` value is shared with child classes.

**Pull request merged**